src-ts/lib/work-provider/work-functions/work-factory/work.factory.ts

```diff
@@ -73,6 +73,29 @@ function findMetadata(challenge: Challenge, metadataName: ChallengeMetadataName)
     return challenge.metadata?.find((item: ChallengeMetadata) => item.name === metadataName)
 }
 
+function findOpenPhase(challenge: Challenge): ChallengePhase | undefined {
```

Comment:
I really like how you added the logic flow comments -- really makes this understandable. 💯

Reply:
gracias